server/app/jobs/certificate_renew_job.rb

```diff
@@ -0,0 +1,88 @@
+class CertificateRenewJob
+  include Celluloid
+  include Logging
+  include CurrentLeader
+  include WaitHelper
+
+  RENEW_INTERVAL = 1.day.to_i
+
+  RENEW_TRESHOLD = 7.days
+
+  def initialize(perform = true)
+    async.perform if perform
+  end
+
+  def perform
+    # TODO Maybe sleep a while first to allow everything else to start up before running this loop?
```

Comment:
Hmm... the way that Celluloid `every` works is that it only runs the block after the interval has expired, not right at the start.

Wondering if the `RENEW_THRESHOLD = 1.day` is now even too long... it means that if the server restarts every night, it might never actually run this job.

Reply:
I think I'd stick to maximum delays on the order of an hour for the `every` loops, so maybe drop the `RENEW_INTERVAL` down to like `1h`?

Longer intervals would really need some kind of scheduling system that manages the timestamps in mongo.

Reply:
oh, very true. :) Maybe something like:
```
sleep 5.minutes
loop do
  do_the_magic
  sleep RENEW_THRESHOLD
end
````
??